.gitlab-ci.yml

```diff
@@ -105,9 +107,25 @@ build-wasm-spiritnet:
     - export AUTHORIZE_UPGRADE_PREFIX=0x02
     - cp -r * /build
     - /srtool/build build
+    - subwasm --json meta /out/spiritnet_runtime.wasm > /out/spiritnet-metadata.json
     - mkdir ./out
     - mv /out/* ./out/
   artifacts:
     paths:
       - out/*.wasm
     expire_in: 12 week
+
+upload:
+  stage: upload
+  dependencies:
+    - build-wasm-spiritnet
+    - build-wasm-peregrine
+  image: amazon/aws-cli:2.9.18
+  script:
+    - aws configure set region $AWS_DEFAULT_REGION
+    - aws s3 cp /out/peregrine-metadata.json s3://$S3_BUCKET/peregrine/${CI_COMMIT_TAG}/
```

Comment:
@ggera as far as I know, `CI_COMMIT_TAG` could be empty if the job was simply triggered by a commit push. Would it make sense to have the same logic we have for Docker image tagging? Or maybe push something like `<branch-commit>` if a regular push and `<tag>` if a tag.